libraries/standard/mqtt/src/iot_mqtt_helper.c

```diff
@@ -859,12 +863,8 @@ IotMqttError_t _IotMqtt_ProcessIncomingPublishFlags( uint8_t publishFlags,
 {
     IotMqttError_t status = IOT_MQTT_SUCCESS;
 
-    if( pOutput == NULL )
-    {
-        status = IOT_MQTT_BAD_RESPONSE;
-    }
     /* Check for QoS 2. */
-    else if( UINT8_CHECK_BIT( publishFlags, MQTT_PUBLISH_FLAG_QOS2 ) == true )
+    if( UINT8_CHECK_BIT( publishFlags, MQTT_PUBLISH_FLAG_QOS2 ) == true )
```

Comment:
See both iot_mqtt_lightweight_api.c 533 and iot_mqtt_serialize.c 541:

```
IotMqttPublishInfo_t * pOutput = &( pPublish->pubInfo );
```

As pOutput is an address of a struct member, it can never be NULL.

Reply:
Correct, however `_IotMqtt_ProcessIncomingPublishFlags()` [isn't declared `static`](https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/v4_beta/libraries/standard/mqtt/src/private/iot_mqtt_helper.h#L280) and could be called from outside of this translation unit.  Consider the case where a future developer adds some buggy code which calls this.  It looks like this pattern of not validating pointer arguments [exists in other places in this file as well](https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/v4_beta/libraries/standard/mqtt/src/iot_mqtt_helper.c#L231), so perhaps this was a deliberate design decision.

Reply:
This was originally a static function.  But now with the lightweight API, it needs to be shared across 2 files (lightweight and full APIs) so we made it non-static.

It's still considered internal though, its signature is only provided by including the internal header.